test/optim/test_optimize.py

```diff
@@ -1551,7 +1586,9 @@ def test_optimize_acqf_mixed_q2(self, mock_optimize_acqf):
             self.assertTrue(torch.equal(acq_value, expected_acq_value))
 
     def test_optimize_acqf_mixed_empty_ff(self):
-        with self.assertRaises(ValueError):
+        with self.assertRaises(
```

Comment:
```suggestion
        with self.assertRaisesRegex(
```
`assertRaises` with `msg` doesn't actually check message equality -- the msg keyword is for the error message for a failure. See here: https://docs.python.org/3/library/unittest.html#unittest.TestCase.debug